common/datavalues/src/types/serializations/nullable.rs

```diff
@@ -98,4 +100,29 @@ impl TypeSerializer for NullableSerializer {
 
         Ok(Arc::new(data))
     }
+
+    fn serialize_csv<'a>(
+        &self,
+        column: &'a ColumnRef,
+        format: &FormatSettings,
+    ) -> Result<Box<dyn StreamingIterator<Item = [u8]> + 'a>> {
```

Comment:
Maybe it's ok, it's more worthy than tmp alloc.

Reply:
a little worried about it too, not have much experience of it,  let me do a simple bench for comparing first

Reply:
https://gist.github.com/youngsofun/75c6420bf51fe4c63e19aa303d2e4f9d

Reply:
Since we can't avoid virtual call, how about using row based API?

Refer to [clickhouse](https://github.com/ClickHouse/ClickHouse/blob/dbc2f3408ea37f4e7ff3ddc35da47dfbaf090f89/src/DataTypes/Serializations/ISerialization.h)

Reply:
Need another bench for the performance.

Reply:
@sundy-li ok, I`ll try it

Reply:
@sundy-li 
the output of the current bench:
https://gist.github.com/youngsofun/b7a2d3629e53aa6d6cdfb72d82a8a727

` fn write_csv_field(&self,  column, row_num, &mut buf)`  is faster then stream_iterator for short columns, but a little slow for long ones. I think both are ok.

only test int32 today,  tided up the cold,  will go on with nullable and string tomorrow.
stream_iterator needs an extra copy from tmp buf inside iterator, may slow down when the string is long, but may not be too bad if it is in cache?